### PR TITLE
@alloy => [Publishing] Email signup

### PR DESCRIPTION
**Pull request by kanaabe**
Success:
emailsignupsuccess

Error:
emailsignuperror

Todo
~~- [ ] Fix babel-plugin-rewire error with TS~~
- [x] Error handling (pending design)
- [x] Thank you message (pending design)

### CODE REVIEW

**Comment by alloy** [comment]
@kanaabe FYI Jest has module mocking builtin, there shouldn’t be a need for rewiring.

**Comment by kanaabe** [comment]
@alloy aha of course, thanks! man jest is nice.

File: src/components/publishing/email_signup.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          <Title>Stay up to date with Artsy Editorial</Title>
+          <Form>
+            <Input
+              type="email"
+              placeholder="Enter Your Email..."
+              onChange={this.onInputChange}
+              value={this.state.message || this.state.value}
+              readOnly={this.state.message.length > 0}
+              isError={this.state.error}
+              isReadOnly={this.state.message.length > 0}
```
Comment (kanaabe) [inline src/components/publishing/email_signup.tsx]: Seems a little confusing here, but we're making use of `value` and `readOnly` to display a message, whether it be a success or error.

**Comment by alloy** [comment]
LGTM and nice tests, I now learned about the fake timers as well 👍

**Comment by alloy** [comment]
Needs a rebase, though.

**Pull request merged**